src/components/item-sidebars/ShowItemSidebar.tsx

```diff
@@ -14,6 +14,12 @@ import Button from '../button/Button'
 import { MdModeEdit } from 'react-icons/md'
 import { LazyLoadImage } from 'react-lazy-load-image-component'
 import 'react-lazy-load-image-component/src/effects/blur.css'
+import { singleCategoryState } from '../../global-state/categoriesState'
+
+const CategoryText = ({ category_id }: any) => {
+    const singleCategory = useRecoilValue(singleCategoryState(category_id))
+    return <div className="text-xl mt-2">{singleCategory.name}</div>
+}
```

Comment:
I feel like this can be refactored. I would just take the CategoryHeading component, have 2 props, one for category_id, and another for heading props, which would be an object which can be spread over Heading component.

Reply:
I will look into that. I actually tried with a HOC component but didn't end up with something that I liked. But I'm not sure that I did it correctly anyway ;).